lib/fp.js

```diff
@@ -54,13 +31,29 @@ const fmap = (fn1, f) => {
 //   funcA - <Function>
 const ap = (fn, funcA) => concat(funcA, (f, callback) => fmap(fn, f)(callback));
 
-asyncChainMethods = { fmap, ap, concat };
+function FpFunction() {}
+
+//  fpChain replacing prototype of the function fn adding functional methods to
+//  it. Also it binds arguments args to the function fn.
+//    fn - <Function> a function which functional methods will be added to
+//    args - <any[]> arguments which will be partially applied to fn.
+const fpChain = (fn, ...args) => {
+  const wrapped = fn.bind(null, ...args);
+  Object.setPrototypeOf(wrapped, FpFunction.prototype);
+  return wrapped;
+};
+
+const FP_FUNCTION_METHODS = [concat, fmap, ap];
```

Comment:
I'd prefer to have it right after `FpFunction` declaration so it will be easier to check what methods it has.